### PR TITLE
Update exodusii package

### PR DESCRIPTION
**Pull request by gsjaardema**
The exodusii package was neglected for several years, but I found that several people and packages were using it which was causing support issues.  The package is now current with the seacas package and will be maintained going foward.

### CODE REVIEW

**Comment by gsjaardema** [comment]
@spackbot fix style

**Review by sethrj — DISMISSED** [review]
Looks great! Thanks for updating it.

File: var/spack/repos/builtin/packages/exodusii/package.py
Comment (johnwparent) [inline var/spack/repos/builtin/packages/exodusii/package.py:153]: The check here should technically be for `msmpi` being the MPI provider, not just the Windows platform.
I realize you're just porting the logic that's already there (to be clear, this is my mistake), so if you'd like to get this PR in as is, I can be responsible for that change.

**Review by johnwparent — DISMISSED** [review]
Builds without issue exactly as before on Windows! LGTM with one comment that is non-blocking.

File: var/spack/repos/builtin/packages/exodusii/package.py
Comment (gsjaardema) [reply var/spack/repos/builtin/packages/exodusii/package.py:153]: If you don't mind, that would be good.  

We should also look at getting shared library support for exodus working.  There are `EXPORT` defines that decorate each function, but I don't think they are ever activated on a Windows build...  I don't have much access or knowledge on Windows, so any help would be appreciated (understand if that is not something you need...)

Comment (gsjaardema) [reply var/spack/repos/builtin/packages/exodusii/package.py:153]: We should also make sure that the `msmpi` change is also done in the seacas package to keep them consistent.

File: var/spack/repos/builtin/packages/exodusii/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/exodusii/package.py:65]: These checksums are no longer valid. 

You could add `url_for_version` method (see https://spack.readthedocs.io/en/latest/packaging_guide.html#writing-a-package-recipe) or add `url` as an argument to the directory.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the homepage, branch, and new version sha256s but cannot confirm the older sha256s.

File: var/spack/repos/builtin/packages/exodusii/package.py
Comment (johnwparent) [reply var/spack/repos/builtin/packages/exodusii/package.py:153]: > I don't have much access or knowledge on Windows, so any help would be appreciated (understand if that is not something you need...)

I'm always happy to help with Windows work! And exodusII is a pretty important package for a number of paraview related tasks, so I'm happy to help. Feel free to reach out on the Spack Slack to iterate on this (same user name).

> We should also make sure that the msmpi change is also done in the seacas package to keep them consistent.

Yes, I've got a local branch that's adding intel-mpi support to Windows, so everywhere the check is for mpi and windows will be refactored. The goal of the effort is actually to support Exodusii + Fortran + MPI on Windows, so Exodusii and Seacas are high on my support list.

 Since there is a good deal of shared CMake/general code between Exodusii, Seacas, and Trilinos, do you think it would make sense to condense the Spack configuration for common options to some centralized parent class?

File: var/spack/repos/builtin/packages/exodusii/package.py
Comment (gsjaardema) [reply var/spack/repos/builtin/packages/exodusii/package.py:65]: I changed back to using the url that fetches zip files instead of tar.gz files and redid all the new version checksums.  I installed a few of the older versions and new versions and I think all checksums are valid.

File: var/spack/repos/builtin/packages/exodusii/package.py
Comment (gsjaardema) [reply var/spack/repos/builtin/packages/exodusii/package.py:153]: I'm fine doing that with Exodusii and Seacas, but I wouldn't want to include Trilinos just because it would make it more difficult to update if I had to get consensus/permission from Trilinos...  Prefer to do it only on the packages I have control over...

**Review by tldahlgren — APPROVED** [review]
Re-confirmed the homepage, branch, commit, and all version sha256s.

File: var/spack/repos/builtin/packages/exodusii/package.py
Comment (johnwparent) [reply var/spack/repos/builtin/packages/exodusii/package.py:153]: Oh sorry, I saw you maintain Seacas and Exodus and just assumed. 
We can limit to Seacas and Exodus, that probably makes more sense anyway.

**Comment by gsjaardema** [comment]
@tldahlgren @sethrj @johnwparent Can someone with merge permission mrege this if they get a chance.  Thanks.

**Pull request merged**
